perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -351,43 +354,88 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
 
   if metadata is None:
     metadata = {}
-
-  records = json.loads(raw_result)
-  metric_name = 'Multi-stream %s latency' % operation
-
-  metadata = metadata.copy()
   metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
-  overall_metadata = metadata.copy()
+  records_json = json.loads(raw_result)
+  records = pd.DataFrame(records_json)
+
+  all_streams_start, all_streams_duration = analyze.AllStreamsInterval(
+      records['start_time'], records['latency'], records['stream_num'])
+  logging.info(
+      'First stream started %s seconds before last stream started',
+      all_streams_start - records['start_time'].min())
+  logging.info(
+      'Last stream ended %s seconds after first stream ended',
+      (records['start_time'] + records['latency']).max() -
+      (all_streams_start + all_streams_duration))
+  records_in_interval = records[
+      analyze.FullyInInterval(records['start_time'],
+                              records['latency'],
+                              all_streams_start,
+                              all_streams_duration)]
+
   # Don't publish the full distribution in the metadata because doing
   # so might break regexp-based parsers that assume that all metadata
   # values are simple Python objects. However, do add an
   # 'object_size_B' metadata field even for the full results because
   # searching metadata is easier when all records with the same metric
   # name have the same set of metadata fields.
-  overall_metadata['object_size_B'] = 'distribution'
+  distribution_metadata = metadata.copy()
+  distribution_metadata['object_size_B'] = 'distribution'
+
+  latency_prefix = 'Multi-stream %s latency ' % operation
+  throughput_prefix = 'Multi-stream %s throughput ' % operation
   logging.info('Processing %s multi-stream %s results for the full '
                'distribution.', len(records), operation)
-  _AppendPercentilesToResults(
-      results,
-      (record['latency'] for record in records),
-      metric_name,
+  analyze.AppendStatsAsSamples(
+      records_in_interval['latency'],
       'sec',
-      overall_metadata)
+      results,
+      name_prefix=latency_prefix,
+      metadata=distribution_metadata)
 
+  analyze.AppendStatsAsSamples(
+      records_in_interval['size'] / records_in_interval['latency'],
+      'B/sec',
+      results,
+      name_prefix=throughput_prefix,
+      metadata=distribution_metadata)
+
+  logging.info('Processing %s multi-stream %s results for net throughput',
+               len(records), operation)
+  results.append(
+      sample.Sample(
+          'Multi-stream %s net throughput' % operation,
+          records_in_interval['size'].sum() /
+          records_in_interval['latency'].sum(),
```

Comment:
Oh, shoot! That's the wrong calculation. I think my first mistake was doing the net throughput calculation here, where it is difficult to test. So I moved it to analyze.py, added a unit test, and pushed a change that should fix the calculation.

Yes, summing the size of records in records_in_interval equals the total number of bytes transferred during the interval where all streams were active.